src/Microsoft.Crank.Agent/Controllers/JobsController.cs

```diff
@@ -645,7 +651,13 @@ public IActionResult List(int id, string path)
                     return NotFound();
                 }
 
-                var fullPath = Path.Combine(job.BasePath, path);
+                // Resolve dot notation in path
+                var fullPath = Path.GetFullPath(path, job.BasePath);
```

Comment:
Would a tiny helper method be useful here? Looks like you have the very same check in both places.